source/_posts/2020-03-14-2020-using-vyper-with-embark.md

```diff
@@ -0,0 +1,204 @@
+
+Irrespective of the security audit, [Vyper is perhaps the most intuitive and easiest to quickly learn language for reading and writing Etheruem contracts - even without any prior knowledge or experience in programming](https://www.publish0x.com/rhyzom/vyper-compiler-security-audit-and-developing-a-new-rust-base-xxpygm). Additionally, Embark has created a plugin that makes it just as easy to compile Vyper contracts in an Embark &ETH;App!
+
+Let's see just how easy this is!
+
+1. [Prerequisites](#Step-1-Prerequisites)
+2. [Create a &ETH;App](#Step-2-Create-a-ETH-App)
+3. [Install the Embark Vyper plugin](#Step-3-Install-the-Embark-Vyper-plugin)
+4. [Add a Vyper contract to your &ETH;App](#Step-4-Add-a-Vyper-contract-to-your-ETH-App)
+5. [Run Embark](#Step-5-Run-Embark)
+6. [Step-6-Interact-with-your-contract-in-Cockpit](#Step-6-Interact-with-your-contract-in-Cockpit)
+
+## Step 1. Prerequisites
+Before we can write our first Vyper contract, let's take care of a few requirements.
+
+### Install Embark
+[Install Embark](https://framework.embarklabs.io/docs/installation.html) either globally or as a package in your &ETH;App. The most simple way forward is to install Embark globally on your system:
+```
+yarn global add embark
+# OR
+npm i -g embark
+```
+
+The rest of this article will assume you have Embark installed globally, and therefore available from the CLI.
+
+### Install Vyper
+If you haven't already done so, make sure to [install Vyper](https://vyper.readthedocs.io/en/latest/installing-vyper.html). As the documentation recommends, be sure to [use a Python virtual environment when installing Vyper](https://vyper.readthedocs.io/en/latest/installing-vyper.html#creating-a-virtual-environment), as it will keep your system-wide packages from being polluted.
+
+## Step 2. Create a &ETH;App
+For this article, we will be creating a demo &ETH;App to use as a base for creating our first Vyper contract. However, if you already have a &ETH;App that you'd like to add Vyper contracts with, simply skip this step.
+
+Creating an Embark demo is easy, simply run the following commands:
+```
+embark demo
+cd embark_demo
+```
+
+## Step 3. Install the Embark Vyper plugin
+Installing the [Embark Vyper](https://github.com/embarklabs/embark/blob/master/packages/plugins/vyper/README.md) plugin in our &ETH;App is extremely simple:
+1. Add the `embark-vyper` package to your &ETH;App:
+```
+yarn add embark-vyper
+# OR
+npm i embark-vyper
+```
+2. Add the `embark-vyper` plugin to `embark.json`:
+```
+// embark.json
+
+// ...
+"plugins": {
+  "embark-ipfs": {},
+  "embark-swarm": {},
+  "embark-whisper-geth": {},
+  "embark-geth": {},
+  "embark-parity": {},
+  "embark-profiler": {},
+  "embark-graph": {},
+  "embark-vyper": {} // <====== add this!
+},
+// ...
+```
+
+## Step 4. Add a Vyper contract to your &ETH;App
+First, let's delete the `SimpleStorage` Solidity contract that comes with the Embark Demo by default:
+```
+rm -rf contracts/simple_storage.sol
```

Comment:
Maybe better to just do `rm contracts/simple_storage.sol` instead. Reason being:

1. It's just a simple file so `-r` isn't needed
2. If, for whatever reason, the reader types in a wrong path or even just `.`, the `-rf` options can potentially cause big damage.

Omitting those options make `rm` complain when the target is a directory which is another safety net.